addons/binding/org.openhab.binding.nuki/src/main/java/org/openhab/binding/nuki/handler/NukiBridgeHandler.java

```diff
@@ -0,0 +1,104 @@
+import java.util.concurrent.TimeUnit;
+
+import org.eclipse.jetty.http.HttpStatus;
+import org.eclipse.smarthome.core.thing.Bridge;
+import org.eclipse.smarthome.core.thing.ChannelUID;
+import org.eclipse.smarthome.core.thing.ThingStatus;
+import org.eclipse.smarthome.core.thing.ThingStatusDetail;
+import org.eclipse.smarthome.core.thing.binding.BaseBridgeHandler;
+import org.eclipse.smarthome.core.types.Command;
+import org.openhab.binding.nuki.internal.dataexchange.BridgeInfoResponse;
+import org.openhab.binding.nuki.internal.dataexchange.NukiHttpClient;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The {@link NukiBridgeHandler} is responsible for handling commands, which are
+ * sent to one of the channels.
+ *
+ * @author Markus Katter - Initial contribution
+ */
+public class NukiBridgeHandler extends BaseBridgeHandler {
+
+    private final static Logger logger = LoggerFactory.getLogger(NukiBridgeHandler.class);
+    private final static int RETRYJOB_INTERVAL = 60;
+
+    private NukiHttpClient nukiHttpClient;
+    private ScheduledFuture<?> retryJob;
+
+    public NukiBridgeHandler(Bridge bridge) {
+        super(bridge);
+        logger.trace("Instantiating NukiBridgeHandler({})", bridge);
+    }
+
+    public NukiHttpClient getNukiHttpClient() {
+        return nukiHttpClient;
+    }
+
+    @Override
+    public void initialize() {
+        logger.debug("NukiBridgeHandler:initialize()");
+        scheduler.execute(() -> initializeHandler());
+    }
+
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+        logger.debug("NukiBridgeHandler:handleCommand({}, {})", channelUID, command);
+    }
+
+    @Override
+    public void dispose() {
+        logger.debug("NukiBridgeHandler:dispose()");
+        nukiHttpClient.stopClient();
+        nukiHttpClient = null;
+        cancelRetryJob();
+    }
+
+    private void initializeHandler() {
+        logger.debug("NukiBridgeHandler:initializeHandler()");
+        if (nukiHttpClient == null) {
+            nukiHttpClient = new NukiHttpClient(getConfig());
+        }
+        BridgeInfoResponse bridgeInfoResponse = nukiHttpClient.getBridgeInfo();
+        if (bridgeInfoResponse.getStatus() == HttpStatus.OK_200) {
+            updateStatus(ThingStatus.ONLINE);
+            cancelRetryJob();
+        } else {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR, bridgeInfoResponse.getMessage());
+            scheduleRetryJob();
+        }
+    }
+
+    private void scheduleRetryJob() {
+        logger.trace("NukiBridgeHandler:scheduleRetryJob():Scheduling retryJob in {}secs for Bridge.",
+                RETRYJOB_INTERVAL);
+        if (retryJob != null && !retryJob.isDone()) {
+            logger.trace("NukiBridgeHandler:scheduleRetryJob():Already scheduled for Bridge.");
+            return;
+        }
+        retryJob = null;
+        retryJob = scheduler.schedule(() -> {
+            initialize();
```

Comment:
As I am scheduling the retry job in `initializeHandler()` the job would never be `isDone()`. I do not think it is a problem to call `initialize();` here as it actually just calls `initializeHandler()`. If you insist I will have to rework the scheduler. Let me know.

Reply:
> it actually just calls initializeHandler()

Well, it
(1) also sets the status to UNKOWN, which might not be wished here as the status is usually OFFLINE and expected to change to ONLINE only.
(2) schedules the call in another job, so you have a job in a job, which looks like a bad design to me.

> the job would never be isDone()

Well, with a job in the job, it is now immediately done and thus the `isDone()` check does not have any meaning at all.

>  If you insist I will have to rework the scheduler.

I honestly think that this might make sense to come to a simpler and better traceable program flow.
What e.g. speaks against scheduling a regular "checkConnection" job in `initialize()` and simply keep it running without cancelling and re-scheduling it all the time? If could do a quick check whether the connection is established (and send an ONLINE status) and if it is NOT there, go an re-establish the connection.
This would also prevent it from continuously creating new NukiHttpClient client instances, which seems to be unnecessary.